src/Tmds.Ssh/KeyExchange.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tmds.Ssh;
+
+abstract class KeyExchange
```

Comment:
Please put the `IKeyExchangeAlgorithm` here and remove it from the derived classes.